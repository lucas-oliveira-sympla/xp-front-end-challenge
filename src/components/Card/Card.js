import React from 'react';
import { string } from 'prop-types';

import { CardStyled, ImageStyled } from './Card.styles';

const propTypes = {
  imgUrl: string.isRequired,
  album: string.isRequired,
  artist: string.isRequired,
};

export const renderCard = (imgUrl, album, artist, bigger) => {
  const alt = `${album} - ${artist}`;
  return (
    <CardStyled bigger={bigger}>
      <ImageStyled bigger={bigger}>{imgUrl && <img src={imgUrl} alt={alt} />}</ImageStyled>
      <figcaption>
        <span>{album}</span>
        <span>{artist}</span>
      </figcaption>
    </CardStyled>
  );
};

const Card = ({ imgUrl, album, artist }) => renderCard(imgUrl, album, artist);
Card.Bigger = ({ imgUrl, album, artist }) => renderCard(imgUrl, album, artist, true);

Card.propTypes = propTypes;

export default Card;
