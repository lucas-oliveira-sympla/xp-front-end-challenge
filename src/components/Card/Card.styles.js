/* eslint-disable no-confusing-arrow */
import styled from 'styled-components';

const sizes = {
  default: 170,
  bigger: 300,
};

export const CardStyled = styled.figure`
  max-width: ${(props) => (props.bigger ? sizes.bigger : sizes.default)}px;
`;

export const ImageStyled = styled.div`
  background-color: #fafafa;
  display: inline-block;
  height: ${(props) => (props.bigger ? sizes.bigger : sizes.default)}px;
  width: ${(props) => (props.bigger ? sizes.bigger : sizes.default)}px;

  > img {
    width: 100%;
  }
`;

export const AlbumStyled = styled.figcaption`
  color: #fff;
  font-family: 'Roboto', sans-serif;
  font-size: 14px;
  line-height: 16px;
  margin: 7px 0;
  text-align: center;
`;

export const ArtistStyled = styled(AlbumStyled)`
  color: #999;
`;
