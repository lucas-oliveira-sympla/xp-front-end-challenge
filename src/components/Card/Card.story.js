import React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';

import { BackgroundStyled } from '../../App.styles';

import Card from './Card';

const SmallWrapper = styled.div`
  width: 170px;
`;

const BiggerWrapper = styled.div`
  width: 300px;
`;

const cardProps = {
  album: 'Nome do album',
  artist: 'Nome do artista',
  imgUrl:
    'https://cdn.smehost.net/rcarecordscom-usrcaprod/wp-content/uploads/2013/05/Run-Cover.jpg',
};

const doubleLineProps = {
  album: 'Nome do album grande de duas linhas',
  artist: 'Nome do artista grande de duas linhas',
  imgUrl:
    'https://cdn.smehost.net/rcarecordscom-usrcaprod/wp-content/uploads/2013/05/Run-Cover.jpg',
};

const stories = storiesOf('Card', module);
stories
  .add('default', () => (
    <BackgroundStyled>
      <SmallWrapper>
        <Card {...cardProps} />
      </SmallWrapper>
    </BackgroundStyled>
  ))
  .add('with image', () => (
    <BackgroundStyled>
      <SmallWrapper>
        <Card {...doubleLineProps} />
      </SmallWrapper>
    </BackgroundStyled>
  ))
  .add('Bigger default', () => (
    <BackgroundStyled>
      <BiggerWrapper>
        <Card.Bigger {...cardProps} />
      </BiggerWrapper>
    </BackgroundStyled>
  ))
  .add('Bigger with image', () => (
    <BackgroundStyled>
      <BiggerWrapper>
        <Card.Bigger {...doubleLineProps} />
      </BiggerWrapper>
    </BackgroundStyled>
  ));
