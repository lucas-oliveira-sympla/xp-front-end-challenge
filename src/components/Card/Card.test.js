import React from 'react';
import { mount } from 'enzyme';

import Card from './Card';

const cardProps = {
  album: 'Nome do album',
  artist: 'Nome do artista',
  imgUrl:
    'https://cdn.smehost.net/rcarecordscom-usrcaprod/wp-content/uploads/2013/05/Run-Cover.jpg',
};

describe('<Card />', () => {
  let wrapper;

  describe('with default props', () => {
    beforeAll(() => {
      wrapper = mount(<Card {...cardProps} />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('with bigger props', () => {
    beforeAll(() => {
      wrapper = mount(<Card.Bigger {...cardProps} />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
