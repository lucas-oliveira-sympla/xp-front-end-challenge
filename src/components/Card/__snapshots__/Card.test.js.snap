// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> with bigger props should match snapshot 1`] = `
<Component
  album="Nome do album"
  artist="Nome do artista"
  imgUrl="https://cdn.smehost.net/rcarecordscom-usrcaprod/wp-content/uploads/2013/05/Run-Cover.jpg"
>
  <Cardstyles__CardStyled
    bigger={true}
  >
    <StyledComponent
      bigger={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Cardstyles__CardStyled-sc-1dpv5pg-0",
            "isStatic": false,
            "lastClassName": "gKqnuC",
            "rules": Array [
              "max-width:100%;margin:0;> figcaption{padding:0 ",
              [Function],
              "px;> span{display:inline-block;width:100%;color:",
              "#FFF",
              ";font-family:'Roboto',sans-serif;text-align:center;margin-top:7px;font-size:14px;line-height:16px;}",
              [Function],
              " > span:nth-child(2){color:",
              "#999",
              ";}}",
            ],
          },
          "displayName": "Cardstyles__CardStyled",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Cardstyles__CardStyled-sc-1dpv5pg-0",
          "target": "figure",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <figure
        className="Cardstyles__CardStyled-sc-1dpv5pg-0 gKqnuC"
      >
        <Cardstyles__ImageStyled
          bigger={true}
        >
          <StyledComponent
            bigger={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Cardstyles__ImageStyled-sc-1dpv5pg-1",
                  "isStatic": false,
                  "lastClassName": "fCDhzC",
                  "rules": Array [
                    "display:inline-block;width:100%;> img{width:100%;}",
                  ],
                },
                "displayName": "Cardstyles__ImageStyled",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Cardstyles__ImageStyled-sc-1dpv5pg-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="Cardstyles__ImageStyled-sc-1dpv5pg-1 fCDhzC"
            >
              <img
                alt="Nome do album - Nome do artista"
                src="https://cdn.smehost.net/rcarecordscom-usrcaprod/wp-content/uploads/2013/05/Run-Cover.jpg"
              />
            </div>
          </StyledComponent>
        </Cardstyles__ImageStyled>
        <figcaption>
          <span>
            Nome do album
          </span>
          <span>
            Nome do artista
          </span>
        </figcaption>
      </figure>
    </StyledComponent>
  </Cardstyles__CardStyled>
</Component>
`;

exports[`<Card /> with default props should match snapshot 1`] = `
<Card
  album="Nome do album"
  artist="Nome do artista"
  imgUrl="https://cdn.smehost.net/rcarecordscom-usrcaprod/wp-content/uploads/2013/05/Run-Cover.jpg"
>
  <Cardstyles__CardStyled>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Cardstyles__CardStyled-sc-1dpv5pg-0",
            "isStatic": false,
            "lastClassName": "gfdXlx",
            "rules": Array [
              "max-width:100%;margin:0;> figcaption{padding:0 ",
              [Function],
              "px;> span{display:inline-block;width:100%;color:",
              "#FFF",
              ";font-family:'Roboto',sans-serif;text-align:center;margin-top:7px;font-size:14px;line-height:16px;}",
              [Function],
              " > span:nth-child(2){color:",
              "#999",
              ";}}",
            ],
          },
          "displayName": "Cardstyles__CardStyled",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Cardstyles__CardStyled-sc-1dpv5pg-0",
          "target": "figure",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <figure
        className="Cardstyles__CardStyled-sc-1dpv5pg-0 gfdXlx"
      >
        <Cardstyles__ImageStyled>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Cardstyles__ImageStyled-sc-1dpv5pg-1",
                  "isStatic": false,
                  "lastClassName": "fCDhzC",
                  "rules": Array [
                    "display:inline-block;width:100%;> img{width:100%;}",
                  ],
                },
                "displayName": "Cardstyles__ImageStyled",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Cardstyles__ImageStyled-sc-1dpv5pg-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="Cardstyles__ImageStyled-sc-1dpv5pg-1 fCDhzC"
            >
              <img
                alt="Nome do album - Nome do artista"
                src="https://cdn.smehost.net/rcarecordscom-usrcaprod/wp-content/uploads/2013/05/Run-Cover.jpg"
              />
            </div>
          </StyledComponent>
        </Cardstyles__ImageStyled>
        <figcaption>
          <span>
            Nome do album
          </span>
          <span>
            Nome do artista
          </span>
        </figcaption>
      </figure>
    </StyledComponent>
  </Cardstyles__CardStyled>
</Card>
`;
